src/nvim/api/window.c

```diff
@@ -59,7 +59,7 @@ void nvim_win_set_buf(Window window, Buffer buffer, Error *err)
 {
   win_T *win = find_window_by_handle(window, err);
   buf_T *buf = find_buffer_by_handle(buffer, err);
-  if (!win || !buf) {
+  if (!win || !buf || win->w_buffer == buf) {
```

Comment:
what happens if we just delete the `w_p_wfb ` check below? `win_set_buf` should already check it. then this problem wouldn't have happened. @ColinKennedy

Reply:
Followup: https://github.com/neovim/neovim/pull/31595